src/data/timer.py

```diff
@@ -42,7 +42,10 @@ def __init__(
 
     @property
     def timer(self) -> 'Timer':
-        return self._timer_ref()
+        timer = self._timer_ref()
+        if timer is None:
+            raise RuntimeError("Timer reference has been garbage collected")
+        return timer
```

Comment:
I don't want to have any trouble with shadowing names
```suggestion
        inner_timer = self._timer_ref()
        if inner_timer is None:
            raise RuntimeError("Timer reference has been garbage collected")
        return inner_timer
```